pkg/services/sqlstore/team.go

```diff
@@ -157,14 +186,18 @@ func SearchTeams(query *models.SearchTeamsQuery) error {
 	var sql bytes.Buffer
 	params := make([]interface{}, 0)
 
+	filteredUsers := getFilteredUsers(query.SignedInUser, query.HiddenUsers)
 	if query.UserIdFilter > 0 {
-		sql.WriteString(getTeamSearchSqlBase())
+		sql.WriteString(getTeamSearchSqlBase(filteredUsers))
 		params = append(params, query.UserIdFilter)
```

Comment:
⚠️ I'm a bit rusty, but it looks to me like you might want to swap the order for adding the user ID filter and the filtered users list? (this is the important comment)

Reply:
Good catch! Thanks!